app/controllers/posts_controller.rb

```diff
@@ -1,12 +1,12 @@
 class PostsController < ApplicationController
-  before_action :find_post, only: %i[show edit update destroy]
+  before_action :find_post, only: %i[index show edit update destroy]
 
   def index
     if params[:user_id]
       @user = User.find(params[:user_id])
-      @posts = @user.posts
+      @posts = @user.posts.paginate(page: params[:page], per_page: 10)
     else
-      @posts = Post.all
+      @posts = Post.paginate(page: params[:page], per_page: 10)
```

Comment:
- [ ] The `will_paginate` does not eliminate N+1 query problems. It only paginates the result of the query. Kindly resolve your N+1 query problem using the `:includes` and remember to update your PR screenshot with before and after images of the solution.